### PR TITLE
Add default metadata for lights entities

### PR DESCRIPTION
**Pull request by helto4real**
## Proposed change

This PR includes default metadata for light domain so we can get default values for attributes like `Brightness` even if the lights are off when generating the first time. This PR also deprecates the usage of `UseAttributeBaseClasses` setting since this chage makes it obsolete and much easier to include additional default metadata for other domains in the future.

## Type of change

- [ ] Dependency upgrade
- [ ] Bugfix (non-breaking change which fixes an issue)
- [x] New feature (which adds functionality to an existing integration)
- [ ] Breaking change (fix/feature causing existing functionality to break)
- [ ] Code quality improvements to existing code or addition of tests

## Additional information

- This PR fixes or closes issue: fixes #
- This PR is related to issue:
- Link to documentation pull request:

## Checklist

- [x] The code change is tested and works locally.
- [x] Local tests pass. **Your PR cannot be merged unless tests pass**
- [x] There is no commented out code in this PR.
- [x] I have followed the [development checklist][dev-checklist]
- [x] The code compiles without warnings (code quality chek)
- [x] Tests have been added to verify that the new code works.

If user exposed functionality or configuration variables are added/changed:

- [ ] Documentation added/updated for [www.home-assistant.io][docs-repository]

[docs-repository]: https://github.com/net-daemon/docs

### CODE REVIEW

File: src/HassModel/NetDaemon.HassModel.CodeGenerator/MetaData/DefaultMetadata/DefaultEntityMetaData.json [hunk 1]
```diff
@@ -0,0 +1,95 @@
+{
+  "domains": [
```
Comment (FrankBakkerNl) [inline src/HassModel/NetDaemon.HassModel.CodeGenerator/MetaData/DefaultMetadata/DefaultEntityMetaData.json:2]: I think now maybe it is simpler to use object / array initilizers iso loading json from file

File: src/HassModel/NetDaemon.HassModel.CodeGenerator/MetaData/DefaultMetadata/DefaultEntityMetaData.json [hunk 2]
```diff
@@ -0,0 +1,95 @@
...
+          "cSharpName": "XyColor",
+          "clrType": "System.Collections.Generic.IReadOnlyList\u00601[System.Double]"
+        },
+        {
+          "jsonName": "entity_id",
+          "cSharpName": "EntityId",
+          "clrType": "System.Collections.Generic.IReadOnlyList\u00601[System.String]"
+        },        
+        {
+          "jsonName": "some_new_attribute",
```
Comment (FrankBakkerNl) [inline src/HassModel/NetDaemon.HassModel.CodeGenerator/MetaData/DefaultMetadata/DefaultEntityMetaData.json:78]: This is a test you forgot to remove?

File: src/HassModel/NetDaemon.HassModel.CodeGenerator/MetaData/DefaultMetadata/DefaultEntityMetaData.json (hunk 1 above)
Comment (FrankBakkerNl) [reply src/HassModel/NetDaemon.HassModel.CodeGenerator/MetaData/DefaultMetadata/DefaultEntityMetaData.json:2]: On the other hand, this makes it easier to copy paste from the files when we need to add stuff

File: src/HassModel/NetDaemon.HassModel.Tests/CodeGenerator/ControllerTest.cs [hunk 3]
```diff
@@ -0,0 +1,21 @@
...
+    {
+        // ARRANGE
+        var controller = new Controller(new CodeGenerationSettings(), new HomeAssistantSettings());
+
+        // ACT
+        var result = await controller.LoadEntitiesMetaDataAsync();
+
+        // ASSERT
+        result.Domains.Count.Should().Be(1);
+        result.Domains.Single(x => x.Domain == "light").Should().NotBeNull();
```
Comment (FrankBakkerNl) [inline src/HassModel/NetDaemon.HassModel.Tests/CodeGenerator/ControllerTest.cs:19]: The most obvious thing to asser is that the Brightness attribute should be present ofc

**Review by FrankBakkerNl — APPROVED** [review]

File: src/HassModel/NetDaemon.HassModel.CodeGenerator/MetaData/DefaultMetadata/DefaultEntityMetaData.json (hunk 1 above)
Comment (helto4real) [reply src/HassModel/NetDaemon.HassModel.CodeGenerator/MetaData/DefaultMetadata/DefaultEntityMetaData.json:2]: > 

Exactly my thought. Will be very simple to add new domains.

File: src/HassModel/NetDaemon.HassModel.CodeGenerator/MetaData/DefaultMetadata/DefaultEntityMetaData.json (hunk 2 above)
Comment (helto4real) [reply src/HassModel/NetDaemon.HassModel.CodeGenerator/MetaData/DefaultMetadata/DefaultEntityMetaData.json:78]: Good find

**Pull request merged**